anndata/tests/test_io_elementwise.py

```diff
@@ -300,3 +300,71 @@ def test_read_zarr_from_group(tmp_path, consolidated):
     with read_func(pth) as z:
         expected = ad.read_zarr(z["table/table"])
     assert_equal(adata, expected)
+
+
+# Test dask sparse array IO above int32 size limit
+
+
+class CSRCallable:
+    """Dummy class to bypass dask checks"""
+
+    def __new__(cls, shape, dtype) -> sparse.csr_matrix:
+        if len(shape) == 0:
+            shape = (0, 0)
+        if len(shape) == 1:
+            shape = (shape[0], 0)
+        elif len(shape) == 2:
+            pass
+        else:
+            raise ValueError(shape)
+        return sparse.csr_matrix(shape, dtype=dtype)
+
+
+def make_dask_chunk(
+    x: ad.experimental.SparseDataset, start: int, end: int
+) -> DaskArray:
+    import dask.array as da
+    from dask import delayed
+
+    def take_slice(x, idx):
+        return x[idx]
+
+    return da.from_delayed(
+        delayed(take_slice)(x, slice(start, end)),
+        dtype=x.dtype,
+        shape=(end - start, x.shape[1]),
+        meta=CSRCallable,
+    )
+
+
+def sparse_dataset_as_dask(x, stride: int):
+    import dask.array as da
+
+    n_chunks, rem = divmod(x.shape[0], stride)
+
+    chunks = []
+    cur_pos = 0
+    for i in range(n_chunks):
+        chunks.append(make_dask_chunk(x, cur_pos, cur_pos + stride))
+        cur_pos += stride
+    if rem:
+        chunks.append(make_dask_chunk(x, cur_pos, x.shape[0]))
+
+    return da.concatenate(chunks, axis=0)
+
+
+def test_write_big_dask():
+    import dask.array as da
+
+    z = zarr.group()
+    # This needs to be something easily compressible to avoid writing a huge file
+    dense_dask = da.ones(
+        (np.iinfo(np.int32).max + 1, 1), chunks=(10_000_000, 1), dtype=bool
+    )
+    sparse_dask = dense_dask.map_blocks(sparse.csr_matrix)
+    write_elem(z, "sparse", sparse_dask)
```

Comment:
@ivirshup Is the speed of this the blocker?

Reply:
I would actually not be opposed to changing the test to just looking at the written type of the `indptr` array (especially if we use `uint64`, as it is the same memory as `int32`).  The fact that the last integer after the maximum is written correctly could be said to be a side effect of this behavior.

Reply:
I think I will just test for dask writing to 64 bit, yes.

> (especially if we use uint64, as it is the same memory as int32)

I have been unsure what you mean by this.